core/src/main/java/org/elasticsearch/gateway/AsyncShardFetch.java

```diff
@@ -195,8 +203,15 @@ protected synchronized void processAsyncFetch(ShardId shardId, List<T> responses
             for (FailedNodeException failure : failures) {
                 logger.trace("{} processing failure {} for [{}]", shardId, failure, type);
                 NodeEntry<T> nodeEntry = cache.get(failure.nodeId());
-                // if the entry is there, and not marked as failed already, process it
-                if (nodeEntry != null && nodeEntry.isFailed() == false) {
+                // if the entry is there, for the right fetching round and not marked as failed already, process it
```

Comment:
I think this comment is in the wrong place now (should be below the `null` check)